### PR TITLE
1007 model api

### LINKED ISSUE #1007: Rewrite the Zeebe Model API
AT:
* fork the Camunda BPMN model API
* build a transformer for the model into the executable model for the Zeebe broker (this is part of the broker NOT the model API)
* improve fluent builder of the BPMN model API (more like the current Zeebe model builder) --> separate issue
* take the YAML workflow parsing from the Zeebe model API (use the BPMN model builder to create the workflow)

Motivation:
* the current Zeebe model Api is hard to extend (e.g. new BPMN symbols)
* Camunda Model API supports all BPMN symbols 
* it has a dependency to runtime models (e.g. Scala dependency for parsing conditions)

### PR DESCRIPTION
**Pull request by ThorbenLindhauer**
Changes:

- Forks the Camunda BPM model API (https://github.com/camunda/camunda-bpmn-model) and replaces our existing model API
- We now have the complete BPMN metamodel
- We have an extensible mechanism for model validation and transformation
- The parsed model and the transformed model are two different things. The former is a part of the model API, the latter a part of the broker. The parsed model is no longer required for execution, so it can be discarded after transformation, which should be more memory-efficient.
- Model API has no more a Scala and Jackson dependency. This has the implication, that some of the validation must be done in the broker (e.g. JSON path query validation). 
- Integrates the Zeebe fluent builders with the existing model API fluent builder. The resulting builder's API style is inconsistent now, which we can tackle in the future.

Notes:

- I think the first two commits (https://github.com/zeebe-io/zeebe/commit/5d6d6c83bcb18ff562a6553bf609af0ceaf34135 and https://github.com/zeebe-io/zeebe/commit/d85ad331229ed7c3b352b4b2d02b55071e9c20b7), which have massive changesets, do not need an in-detail review because they cover code that was already reviewed in the Camunda BPM project.
- I tried my best to separate the changes into coherent commits. It may be that the intermediate states do not compile and that some changes are in the wrong commit.

closes #1007

### CODE REVIEW

File: broker-core/src/test/java/io/zeebe/broker/workflow/WorkflowInstanceFunctionalTest.java [hunk 1]
```diff
@@ -456,6 +456,8 @@ public void shouldSetSourceRecordPositionCorrectOnJoinXor() {
             .endEvent("end")
             .done();
 
+    Bpmn.writeModelToStream(System.out, workflowDefinition);
```
Comment (saig0) [inline broker-core/src/test/java/io/zeebe/broker/workflow/WorkflowInstanceFunctionalTest.java]: Don't print the model no the console.

**Review by saig0 — COMMENTED** [review]
Looks good :+1: Just a few comments.

File: broker-core/src/main/java/io/zeebe/broker/workflow/model/transformation/handler/CreateWorkflowHandler.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  @Override
+  public Class<Process> getType() {
+    return Process.class;
+  }
+
+  @Override
+  public void transform(Process element, TransformContext context) {
+
+    final String id = element.getId();
+    final ExecutableWorkflow workflow = new ExecutableWorkflow(id);
```
[the remark this answers is not in the document]
Comment (ThorbenLindhauer) [reply broker-core/src/main/java/io/zeebe/broker/workflow/model/transformation/handler/CreateWorkflowHandler.java]: => https://github.com/zeebe-io/zeebe/issues/1146

**Review by saig0 — APPROVED** [review]

**Comment by ThorbenLindhauer** [comment]
bors retry

**Pull request merged**